bin/genesis

```diff
@@ -981,6 +987,25 @@ normal_site_metadata() {
 	fi
 	local stemcell_version=$(cat "${DEPLOYMENT_ENV_DIR}/.site/stemcell/version")
 
+	local stemcell_url=""
+	if [[ -f "${DEPLOYMENT_ENV_DIR}/.site/stemcell/url" ]]; then
+		stemcell_url=$(cat "${DEPLOYMENT_ENV_DIR}/.site/stemcell/url")
+	fi
+	local stemcell_sha1=""
+	if [[ -f "${DEPLOYMENT_ENV_DIR}/.site/stemcell/sha1" ]]; then
+		stemcell_sha1=$(cat "${DEPLOYMENT_ENV_DIR}/.site/stemcell/sha1")
+	fi
+	if [[ -z ${stemcell_url} ]]; then
+		if [[ -n ${GENESIS_INDEX} ]]; then
+			echo >&2 "Checking ${GENESIS_INDEX} for correct URL/SHA1 of stemcell ${stemcell_name}/${stemcell_version}"
+			stemcell_url=$(    check_index url     stemcell "${stemcell_name}" ${stemcell_version})
+			stemcell_sha1=$(   check_index sha1    stemcell "${stemcell_name}" ${stemcell_version})
+			stemcell_version=$(check_index version stemcell "${stemcell_name}" ${stemcell_version})
+		fi
+	elif [[ -z ${stemcell_sha1} ]]; then
+		stemcell_sha1=$(curl -Lsk ${stemcell_url} | checksum | awk '{print $1}')
```

Comment:
I don't like this step - seems like it will lead to a bad user experience. I'd rather just error out and tell the person to specify a sha1 themselves. That  way genesis returns quickly, rather than hanging for 5-60 minutes while silently downloading stemcells that could be upwards of a few GB, and potentially exhausting file/memory resources for the download.